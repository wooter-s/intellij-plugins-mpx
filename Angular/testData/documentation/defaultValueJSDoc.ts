import { Component, Input } from '@angular/core';

@Component({
             selector: 'quick-docs-default',
             standalone: true,
             template: `quick-docs-default works`
           })
class QuickDocsDefault {
  /** This is a JSDoc test
   * @default true
   * @internal this is a test for @internal JSDoc tag
   * @deprecated this is a test for @deprecated JSDoc tag
   * */
  @Input() property = true;
}

@Component({
 standalone: true,
 imports: [
   QuickDocsDefault
 ],
 template: `
    <quick-docs-default
      [prop<caret>erty]='false'
    ></quick-docs-default>`
})
class QuickDocsExample {
}