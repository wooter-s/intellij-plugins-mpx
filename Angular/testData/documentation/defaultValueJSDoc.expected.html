<div class='separated'><div class='definition'><pre><span style='color: #000080'>directive input</span> <span style='color: #660e7a'>property</span><span style='color: #000000'>: </span><span style='color: #000000'><span style="color:#000080;">boolean</span></span></pre></div><div class='content'><p>This is a JSDoc test</div><table class='sections'><tr><td valign='top' class='section'><p>Default:</td><td valign='top'><p><code>true</code><tr><td valign='top' class='section'><p>Internal:</td><td valign='top'><p>this is a test for @internal JSDoc tag</td><tr><td valign='top' class='section'><p>Deprecated:</td><td valign='top'><p>this is a test for @deprecated JSDoc tag</td></table></div>
<div class='definition'><pre><span style='color: #000080'>standalone</span> <span style='color: #000080'>component</span> <span style='color: #000000'><span style="">QuickDocsDefault</span></span></pre></div><table class='sections'><tr><td valign='top' class='section'><p>Selectors:</td><td valign='top'><p><pre><span style="color:#000080;">quick-docs-default</span></pre></td></table><table class='sections'><tr><td valign='top' colspan='2'><icon src='JavaScriptPsiIcons.FileTypes.TypeScriptFile'/>&nbsp;defaultValueJSDoc.ts</td></table>